### PR TITLE
add chat template for deepseek coder model

### PR DESCRIPTION
**Pull request by lvhan028**
https://huggingface.co/deepseek-ai/deepseek-coder-33b-instruct/blob/main/tokenizer_config.json

### CODE REVIEW

**Comment by lvhan028** [comment]
@zhulinJulia24 may add https://huggingface.co/deepseek-ai/deepseek-coder-6.7b-instruct into test cases

File: lmdeploy/model.py [hunk 1]
```diff
@@ -869,6 +869,45 @@ def match(cls, model_path: str) -> Optional[str]:
...
+
+    def __init__(
+            self,
+            session_len=4096,
+            system='',
+            meta_instruction="""You are an AI programming assistant, utilizing the Deepseek Coder model, developed by Deepseek Company, and you only answer questions related to computer science. For politically sensitive questions, security and privacy issues, and other non-computer science questions, you will refuse to answer\n""",  # noqa: E501
+            eosys='',
+            user='### Instruction:\n',
+            eoh='\n',
+            assistant='### Response:\n',
```
Comment (AllentDan) [inline lmdeploy/model.py:884]: Seems lack `eoa` according to https://huggingface.co/deepseek-ai/deepseek-coder-6.7b-instruct/blob/main/tokenizer_config.json#L34

**Review by AllentDan — APPROVED** [review]
LGTM

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**